magma_access_gateway_post_install/agw_post_install.py

```diff
@@ -0,0 +1,340 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import os
+import time
+from subprocess import DEVNULL, call, check_call
+
+import yaml
+from ping3 import ping  # type: ignore[import]
+from systemd import journal  # type: ignore[import]
+
+logger = logging.getLogger("magma_access_gateway_post_install")
+
+
+class AGWPostInstallChecks:
+
+    MAGMA_AGW_INTERFACES = ["eth0", "eth1", "gtp_br0", "uplink_br0"]
+    MAGMA_AGW_SERVICES = [
+        "magma@control_proxy",
+        "magma@directoryd",
+        "magma@dnsd",
+        "magma@enodebd",
+        "magma@magmad",
+        "magma@mme",
+        "magma@mobilityd",
+        "magma@pipelined",
+        "magma@policydb",
+        "magma@redis",
+        "magma@sessiond",
+        "magma@state",
+        "magma@subscriberdb",
+    ]
+    NON_MAGMA_SERVICES = ["sctpd"]
+    REQUIRED_PACKAGES = [
+        "magma",
+        "magma-cpp-redis",
+        "magma-libfluid",
+        "libopenvswitch",
+        "openvswitch-datapath-dkms",
+        "openvswitch-common",
+        "openvswitch-switch",
+    ]
+    REQUIRED_CONTROL_PROXY_KEYS = [
+        "cloud_address",
+        "cloud_port",
+        "bootstrap_address",
+        "bootstrap_port",
+        "rootca_cert",
+        "fluentd_address",
+        "fluentd_port",
+        "sentry_url_python",
+    ]
+    GOT_CLOUD_HEARTBEAT_MSG = "[SyncRPC] Got heartBeat from cloud"
+    CLOUD_CHECKIN_SUCCESSFUL_MSG = "Checkin Successful! Successfully sent states to the cloud!"
+    TIMEOUT_WAITING_FOR_SERVICE = 60
+    WAIT_FOR_SERVICE_INTERVAL = 10
+
+    def __init__(self, root_ca_pem_path: str):
+        self.root_ca_pem_path = root_ca_pem_path
+
+    def prepare_system_for_post_install_checks(self):
+        """Makes sure system is in the right state for performing post-install checks."""
+        self._stop_service("magma@*")
+        self._bring_down_interface("gtp_br0")
+        self._bring_down_interface("uplink_br0")
+        self._restart_service("openvswitch-switch")
+        self._bring_up_interface("gtp_br0")
+        self._bring_up_interface("uplink_br0")
+
+    def check_whether_required_interfaces_are_configured(self):
```

Comment:
Could these methods simply be properties returning true/false and `main` raises an exception when one of them returns False?

Reply:
This comment can be ignored following our conversation